lib/friends/introvert.rb

```diff
@@ -777,5 +779,52 @@ def thing_with_name_in(type, text)
     def bad_line(expected, line_num)
       raise FriendsError, "Expected \"#{expected}\" on line #{line_num}"
     end
+
+    def decide_default_location_output(activity)
+      @activities = stable_sort(@activities)
+      existing_activities = @activities - [activity]
+
+      str = "Default location"
+      if activity != @activities.first
+        str += " from #{earliest_consecutive_default_location_activity_date(activity, existing_activities)} to"
```

Comment:
I like your implementation - definitely way easier to read. I thought using TDD invariably would lead me to your implementation- a learning point for me!

I've swapped out my code for yours, and yup the tests all pass 🙌 I've removed one bit from your code: `< later_activities.last.date` - it was in the line for deciding if an added activity was the latest or not. `!later_activities.empty?` already does thsi check, and the tests still pass. Let me know if it was in for another reason!

Reply:
Ah, good catch—I think you're right that that part of the condition was superfluous!